Assets/MixedRealityToolkit.Services/InputSimulation/Implementation/MouseRotationProvider.cs

```diff
@@ -0,0 +1,147 @@
+                {
+                    // if mousebutton is either left, right or middle
+                    SetWantsMouseJumping(true);
+                }
+                else if (rotationKey.BindingType == KeyBinding.KeyType.Key)
+                {
+                    // if mousebutton is either control, shift or focused
+                    UnityEngine.Cursor.lockState = CursorLockMode.Locked;
+                    // save current cursor visibility before hiding it
+                    wasCursorVisible = UnityEngine.Cursor.visible;
+                    UnityEngine.Cursor.visible = false;
+                }
+
+                // do nothing if (this.MouseLookButton == MouseButton.None)
+            }
+
+            ++numRotating;
+        }
+
+        private static void OnEndRotating(KeyBinding rotationKey)
+        {
+            --numRotating;
+
+            if (numRotating == 0)
+            {
+                if (rotationKey.BindingType == KeyBinding.KeyType.Mouse)
+                {
+                    // if mousebutton is either left, right or middle
+                    SetWantsMouseJumping(false);
+                }
+                else if (rotationKey.BindingType == KeyBinding.KeyType.Key)
+                {
+                    // if mousebutton is either control, shift or focused
+                    UnityEngine.Cursor.lockState = CursorLockMode.None;
+                    UnityEngine.Cursor.visible = wasCursorVisible;
+                }
+
+                // do nothing if (this.MouseLookButton == MouseButton.None)
+            }
+        }
+
+        /// <summary>
+        /// Mouse jumping is where the mouse cursor appears outside the Unity game window, but
+        /// disappears when it enters the Unity game window.
+        /// </summary>
+        /// <param name="wantsJumping">Show the cursor</param>
+        private static void SetWantsMouseJumping(bool wantsJumping)
+        {
+            if (wantsJumping != isMouseJumping)
+            {
+                isMouseJumping = wantsJumping;
+
+                if (wantsJumping)
+                {
+                    // unlock the cursor if it was locked
+                    UnityEngine.Cursor.lockState = CursorLockMode.None;
+
+                    // save original state of cursor before hiding
+                    wasCursorVisible = UnityEngine.Cursor.visible;
+                    // hide the cursor
+                    UnityEngine.Cursor.visible = false;
+                }
+                else
+                {
+                    // recenter the cursor (setting lockCursor has side-effects under the hood)
+                    UnityEngine.Cursor.lockState = CursorLockMode.Locked;
+                    UnityEngine.Cursor.lockState = CursorLockMode.None;
+
+                    // restore the cursor
+                    UnityEngine.Cursor.visible = wasCursorVisible;
+                }
+
+    #if UNITY_EDITOR
```

Comment:
>#if UNITY_EDITOR [](start = 4, length = 16)

This asmdef looks to be editor-only - I'm curious why UNITY_EDITOR is specicially needed here (would be worth adding a comment here for that reason)

Reply:
1. This code originated in ManualCameraControl, which was previously in the all-platforms Microsoft.MixedReality.Toolkit.Services.InputSimulation asmdef. Since it's only used by the input simulation implementation class i have moved it into the Editor subfolder.
1. The Editor subfolder contains both genuine editor code (inspectors, windows) as well as implementation classes that just happen to be editor-only at this point. In the future it's possible that these implementation classes will be enabled for all platforms, so i would prefer to keep this #ifdef.